cmd/libsnap-confine-private/snap-dir.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#define _GNU_SOURCE
+
+#include "snap-dir.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "snap.h"  // For SC_SNAP_MOUNT_DIR_UNSUPPORTED
+
+static const char *_snap_mount_dir = NULL;
+
+const char *sc_snap_mount_dir(sc_error **errorp) {
+    sc_error *err = NULL;
+
+    if (_snap_mount_dir == NULL) {
+        err = sc_error_init_api_misuse("sc_probe_snap_mount_dir was not called yet");
```

Comment:
since this returns an error anyway, can't we probe here as well? this way you don't really have to think of calling probe early in the program and just go straight to to make the mount dir call

Reply:
There's only one place that probes so I though it's safe to split this. I wanted to probe in the place where we move to the right mount namespace but then I realized we have /proc/$PID/root.